cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -628,21 +618,22 @@ func (h *bundleHandler) addCharm(change *bundlechanges.AddCharmChange) error {
 	}
 
 	// A channel is needed whether the risk is valid or not.
-	var channel charm.Channel
-	if charm.CharmHub.Matches(ch.Schema) {
-		channel = corecharm.DefaultChannel
-		if chParams.Channel != "" {
-			channel, err = charm.ParseChannelNormalize(chParams.Channel)
-			if err != nil {
-				return errors.Trace(err)
-			}
+	channel := corecharm.DefaultChannel
+	if chParams.Channel != "" {
+		channel, err = charm.ParseChannelNormalize(chParams.Channel)
+		if err != nil {
+			return errors.Trace(err)
 		}
-	} else {
-		channel = corecharm.MakeRiskOnlyChannel(chParams.Channel)
```

Comment:
Can we finally delete `MakeRiskOnlyChannel`

Reply:
Yes we can!